src/main/java/ch/njol/skript/expressions/ExprSets.java

```diff
@@ -39,152 +29,74 @@
 import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
-import ch.njol.skript.util.Color;
-import ch.njol.skript.util.SkriptColor;
 import ch.njol.util.Kleenean;
-import ch.njol.util.NullableChecker;
-import ch.njol.util.coll.iterator.ArrayIterator;
-import ch.njol.util.coll.iterator.CheckedIterator;
-import ch.njol.util.coll.iterator.IteratorIterable;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Supplier;
 
 @Name("Sets")
-@Description("Collection sets of items or blocks of a specific type or colours, useful for looping.")
+@Description("A list of all the values of a type; useful for looping.")
 @Examples({
-		"loop items of type ore and log:",
-				"\tblock contains loop-item",
-				"\tmessage \"Theres at least one %loop-item% in this block\"",
-		"drop all blocks at the player # drops one of every block at the player"
+	"loop all attribute types:",
+	"\tset loop-value attribute of player to 10",
+	"\tmessage \"Set attribute %loop-value% to 10!\""
 })
-@Since("<i>unknown</i> (before 1.4.2), INSERT VERSION (colors)")
-public class ExprSets extends SimpleExpression<Object> {
+@Since("INSERT VERSION")
+public class ExprSets<T> extends SimpleExpression<T> {
 
 	static {
 		Skript.registerExpression(ExprSets.class, Object.class, ExpressionType.COMBINED,
-				"[(all [[of] the]|the|every)] item(s|[ ]types)", "[(all [[of] the]|the)] items of type[s] %itemtypes%",
-				"[(all [[of] the]|the|every)] block(s|[ ]types)", "[(all [[of] the]|the)] blocks of type[s] %itemtypes%",
-				"([all [[of] the]] colo[u]rs|(the|every) colo[u]r)");
```

Comment:
We can't be removing these syntaxes without an exact replacement. This will break users scripts.

Reply:
the exact replacement is `[(all [[of] the]|the|every)] %*classinfo%`, all the previous syntaxes are valid with this syntax